### PR TITLE
MM-39856 Update Cloud Plugin dependencies

### PR DESCRIPTION
**Pull request by gigawhitlocks**
#### Summary

This PR updates a whole bunch of dependencies in the Cloud Plugin so that MM-39779 can be addressed, as the resolution of MM-39779 requires the use of an updated Cloud client in order to use mmctl. 

During this update process I also discovered that the webapp portion of the Cloud Plugin was failing to build at all, and this change also updates the dependencies of that portion of the Plugin in order to allow the build to succeed.

Call sites of these dependencies are updated as well in order to allow the build to succeed. 

Also a shout-out to @neallred for helping me get out of dependency Hell in JS-land. Thanks again!

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-39856

#### Release Note

```release-note
Updated Cloud Plugin dependencies to be in line with the Mattermost 6.0 release
```

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -13,20 +13,21 @@
     "test-ci": "jest --forceExit --detectOpenHandles --maxWorkers=2"
   },
   "dependencies": {
+    "core-js": "^3.19.1",
```
Comment (neallred) [inline webapp/package.json:16]: We shouldn't have core-js as a direct dependency. Could we `npm uninstall it`?

I'm guessing adding was partly an attempt to get babel to upgrade the version it needs and pulls in. It looks like babel's latest version is 7.16.0. Could we try upgrading `@babel/cli` and all the `@babel/plugin-*` packages to 7.16.0 (the babel plugin packages usually move lockstep in version with babel)?

**Comment by neallred** [comment]
Nice work on the JS changes! It takes persistence to slog through all the required changes to bring something up to date like this.

File: webapp/package.json (hunk 1 above)
Comment (neallred) [reply webapp/package.json:17]: I stand corrected. This _is_ the right approach: https://babeljs.io/docs/en/babel-polyfill

**Review by neallred — APPROVED** [review]
lgtm!

File: webapp/package.json (hunk 1 above)
Comment (gigawhitlocks) [reply webapp/package.json:17]: Approach followed in commit b599688

**Review by Szymongib — APPROVED** [review]
LGTM

File: build/deploy/main.go [hunk 2]
```diff
@@ -83,15 +83,15 @@ func uploadPlugin(client *model.Client4, pluginID, bundlePath string) error {
 	defer pluginBundle.Close()
 
 	log.Print("Uploading plugin via API.")
-	_, resp := client.UploadPluginForced(pluginBundle)
-	if resp.Error != nil {
-		return fmt.Errorf("Failed to upload plugin bundle: %s", resp.Error.Error())
+	_, _, err = client.UploadPluginForced(pluginBundle)
+	if err != nil {
+		return fmt.Errorf("Failed to upload plugin bundle: %s", err.Error())
```
Comment (gabrieljackson) [inline build/deploy/main.go:88]: nit: `errors.Wrapf()` here would be just a bit cleaner.

File: build/deploy/main.go [hunk 3]
```diff
@@ -83,15 +83,15 @@ func uploadPlugin(client *model.Client4, pluginID, bundlePath string) error {
...
 	}
 
 	log.Print("Enabling plugin.")
-	_, resp = client.EnablePlugin(pluginID)
-	if resp.Error != nil {
-		return fmt.Errorf("Failed to enable plugin: %s", resp.Error.Error())
+	_, err = client.EnablePlugin(pluginID)
+	if err != nil {
+		return fmt.Errorf("Failed to enable plugin: %s", err.Error())
```
Comment (gabrieljackson) [inline build/deploy/main.go:94]: same here

**Review by gabrieljackson — APPROVED** [review]
Nice!

File: build/deploy/main.go (hunk 2 above)
Comment (gigawhitlocks) [reply build/deploy/main.go:88]: ah yeah, you can see how I was just kind of doing a dumb translation to correct the number of return values and argument types and didn't really think about what was actually being called to create the error, here

**Pull request merged**